src/libostree/ostree-repo.c

```diff
@@ -2693,8 +2693,10 @@ min_free_space_size_validate_and_convert (OstreeRepo    *self,
 
   guint64 min_free_space = g_ascii_strtoull (size_str, NULL, 10);
   if (shifts > 0 && g_bit_nth_lsf (min_free_space, 63 - shifts) != -1)
-    return glnx_throw (error, "Integer overflow detected");
-
+  {
+    g_warning ("min-free-space-size: Integer overflow detected");
```

Comment:
Why add the `g_warning()`? The error is already being reported to the caller.

Reply:
Shouldn't be added? I thought its a benefit to the programmer to notice a warning(and infer exact problem) rather than to see a "Value too high". Maybe I am wrong :thinking:

Reply:
The problem is already being reported using the `GError`. When this code is called by a UI, the `g_warning()` will (typically) go to the systemd journal, which the user of the UI would probably never see. When this code is called from a console application, that application is probably already doing its own printing out of the `GError` to `stderr`, so you’d end up with the same error being printed twice.

More generally, using `g_warning()` makes it hard to unit test code, since it will typically be escalated into an `abort()` by `G_DEBUG=fatal-warnings`.

Reply:
hmm, Ok.
Reworked the PR.